input-profile.js

```diff
@@ -0,0 +1,249 @@
+import {Component, Property} from '@wonderlandengine/api';
+import {vec3, quat} from 'gl-matrix';
+const _TempVec = vec3.create();
+const _TempQuat = quat.create();
+
+export class InputProfile extends Component {
+    static TypeName = 'input-profile';
+    static Properties = {
+        controller: Property.object(),
+        handednessIndex: Property.enum(['left', 'right'], 0),
+        path: Property.string(
+            'https://cdn.jsdelivr.net/npm/@webxr-input-profiles/assets@1.0.13/dist/profiles/'
+        ),
+        customProfileFolder: Property.string(''),
+        preSpawnedController: Property.object(),
+    };
+
+    init() {
+        this._gamepadObjects = {};
+        this._modelLoaded = false;
+        this.controllerDeactivateFlag = false;
+        this.ProfileJSON = {};
+    }
+
+    start() {
+        if (this.engine.XRSession) {
+            this.engine.XRSession.addEventListener(
+                'inputsourceschange',
+                this.onInputSourcesChange.bind(this)
+            );
+        } else {
+            this.engine.onXRSessionStart.add((session) => {
```

Comment:
You don't need to do the `if`, because `onXRSessionStart` is a retained emitter, it keeps its values. Please try it in case there is a bug but there shouldn't be any

Reply:
True, thanks , changed :pray:

Reply:
As I understood back in the time ,